tests/unit/server/api/v1/test_datasets.py

```diff
@@ -4802,11 +4802,10 @@ async def test_update_dataset_records(
         assert records[0].suggestions[1].value == "suggestion updated 0 2"
         assert records[0].suggestions[2].value == "suggestion updated 0 3"
         for vector in records[0].vectors:
-            assert records_0_updated_vectors[vector.vector_settings.name] == vector.value
+            if vector.vector_settings.name in records_0_updated_vectors:
```

Comment:
It's a bit hard to understand completely this test. Maybe it makes sense to separate into different simpler tests, checking each of the updatable entities (update_suggestions, update_vectors, update_metadata,....) 

I feel as watching a David Lynch film, concretely, Eraserhead 😅

Reply:
agree, I'll split the test